src/Compilers/Core/Portable/Syntax/SyntaxTriviaList.cs

```diff
@@ -243,26 +244,39 @@ public SyntaxTriviaList Insert(int index, SyntaxTrivia trivia)
         /// <param name="trivia">The trivia to insert.</param>
         public SyntaxTriviaList InsertRange(int index, IEnumerable<SyntaxTrivia> trivia)
         {
-            if (index < 0 || index > this.Count)
+            var thisCount = this.Count;
+            if (index < 0 || index > thisCount)
             {
                 throw new ArgumentOutOfRangeException(nameof(index));
             }
 
-            var items = trivia.ToList();
-            if (items.Count == 0)
+            if (trivia == null)
+            {
+                throw new ArgumentNullException(nameof(trivia));
+            }
+
+            // Handle common case where we're passed an empty array.
+            var triviaList = trivia as IList<SyntaxTrivia>;
```

Comment:
>IList<SyntaxTrivia> [](start = 39, length = 19)

Consider casting to ICollection<T> instead, I think it is the actual owner for the Count property.